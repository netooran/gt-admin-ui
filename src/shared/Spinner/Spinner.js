import { Loader } from '@styled-icons/boxicons-regular';
import styled from 'styled-components';
import { Center } from '../Containers/Containers';

const Wrapper = styled(Center)`
  @keyframes spin {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }

  animation: spin 2s linear infinite;

  svg {
    width: 3rem;
  }
`;

const Spinner = () => {
  return (
    <Wrapper aria-label="Loading">
      <Loader />
    </Wrapper>
  );
};

export default Spinner;
