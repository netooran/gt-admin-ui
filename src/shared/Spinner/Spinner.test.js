import { render } from '@testing-library/react';
import Spinner from './Spinner';

test('renders spinner', () => {
  const { container } = render(<Spinner />);
  expect(container).toMatchSnapshot();
});
