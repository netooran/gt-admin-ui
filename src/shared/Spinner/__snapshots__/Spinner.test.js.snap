// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders spinner 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  display: grid;
  place-items: center;
  width: 100%;
  height: 100%;
}

.c1 {
  -webkit-animation: spin 2s linear infinite;
  animation: spin 2s linear infinite;
}

.c1 svg {
  width: 3rem;
}

<div>
  <div
    aria-label="Loading"
    class="c0 c1"
  >
    <svg
      aria-hidden="true"
      class="c2"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2 11h5v2H2zm15 0h5v2h-5zm-6 6h2v5h-2zm0-15h2v5h-2zM4.222 5.636l1.414-1.414 3.536 3.536-1.414 1.414zM19.778 18.364l-1.414 1.414-3.536-3.536 1.414-1.414zM7.758 14.828l1.414 1.414-3.536 3.536-1.414-1.414zM14.828 7.757l3.536-3.535 1.414 1.415-3.536 3.535z"
      />
    </svg>
  </div>
</div>
`;
